app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -35,12 +75,18 @@ open class FeedViewModel : ViewModel() {
  * It's not necessary to use this factory at this stage. But if we will need to inject
  * dependencies into [FeedViewModel] in the future, then this is the place to do it.
  */
-class FeedViewModelFactory : ViewModelProvider.Factory {
-    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
+class FeedViewModelFactory(context: Context) : ViewModelProvider.Factory {
+
+    private val db = Room.databaseBuilder(
```

Comment:
Can you think of a way to make the database a `Singleton` too? 
This is slightly more tricky than what we did with  the API service, because you need to provide the `Context`.

Reply:
I thought of two suggestions:
1. Create an `FeedApp : Application` class and build the db inside it.
2. Make the database a `Singleton`, and build it only at the first time it's being accessed, using the first context it gets:
`companion object {
        private var db: FeedDatabase? = null
    }

    init {
        if (db == null) {
            db = Room.databaseBuilder(
                context,
                FeedDatabase::
                class.java, "feed_data_base"
            ).build()
        }
    }`

But that might be a bit of a problem in multithreaded access situation, which in the worst case will result with building the same database more than once.